Source/WTF/wtf/StdLibExtras.h

```diff
@@ -589,8 +589,23 @@ ALWAYS_INLINE constexpr typename remove_reference<T>::type&& move(T&& value)
 
 namespace WTF {
 
+template<typename T> class TypeHasRefMemberFunction {
+    template<typename> static std::false_type test(...);
+    template<typename U> static auto test(int) -> decltype(std::declval<U>().ref(), std::true_type());
+public:
+    static constexpr bool value = std::is_same<decltype(test<T>(0)), std::true_type>::value;
+};
+
 template<class T, class... Args>
 ALWAYS_INLINE decltype(auto) makeUnique(Args&&... args)
+{
+    static_assert(std::is_same<typename T::webkitFastMalloced, int>::value, "T is FastMalloced");
+    static_assert(!TypeHasRefMemberFunction<T>::value, "T should not be refcounted");
```

Comment:
In wtf/TypeTraits.h you have `HasRefCountMethods` which does the same as the above (thought it also checks for the presence of `deref()`